deploy/lib/server_config.rb

```diff
@@ -1971,6 +1973,54 @@ def deploy_src
     end
   end
 
+  def deploy_config
+    xquery_dir = @properties['ml.xquery.dir']
+    app_configs = @properties['ml.application-conf-file']
+    total_count = 0
+
+    modules_databases.each do |dest_db|
+      if dest_db == "filesystem"
+        logger.info "Skipping deployment of configurations to #{dest_db}.."
+        break
+      end
+
+      src_permissions = permissions(@properties['ml.app-role'], Roxy::ContentCapability::ERU)
+
+      if ['rest', 'hybrid'].include? @properties["ml.app-type"]
+        # This app uses the REST API, so grant permissions to the rest roles. This allows REST extensions to call
+        # modules not deployed through the REST API.
+        # These roles are present in MarkLogic 6+.
+        src_permissions.push permissions('rest-admin', Roxy::ContentCapability::RU)
+        src_permissions.push permissions('rest-extension-user', Roxy::ContentCapability::EXECUTE)
+        src_permissions.flatten!
+      end
+
+      @logger.debug("source permissions: #{src_permissions}")
+
+      if app_configs.present?
+        logger.info "Deploying application configurations"
+        app_configs.split(',').each do |item|
+          buffer = File.read item
+          replace_properties(buffer, File.basename(item))
+          item_name = item
+          if item.start_with?("src/")
+            item_name = item[4, item.length]
+          end
+          item_name = '/' + item_name
+          logger.info "deploying application configuration #{item} with name #{item_name} on #{dest_db}"
+          total_count += xcc.load_buffer item_name,
+                                         buffer,
+                                         :db => dest_db,
+                                         :add_prefix => File.join(@properties["ml.modules-root"], "app/config/"),
+                                         :permissions => src_permissions
+        end
+        logger.info "Done deploying application configurations"
+      end
+
+      logger.info "\nLoaded #{total_count} #{pluralize(total_count, "document", "documents")} from #{xquery_dir} to #{xcc.hostname}:#{xcc.port}/#{dest_db} at #{DateTime.now.strftime('%m/%d/%Y %I:%M:%S %P')}\n"
+    end
+  end
+
```

Comment:
This function repeats a lot of code from deploy_src. Why isn't this just a modification to lines [1934-1936](https://github.com/marklogic/roxy/blob/dev/deploy/lib/server_config.rb#L1934-L1936) and [1945-1947](https://github.com/marklogic/roxy/blob/dev/deploy/lib/server_config.rb#L1945-L1947)? 

The existing deploy_src function still has the same functionality as before, which means duplicated processing. I'd either drop or reduce this function and modify the code in deploy_src, perhaps to call a smaller version of this function.